### ISSUE TITLE
[DisplayAsProperty] attribute

### ISSUE THREAD

**Issue by richardpawson** [NakedObjectsGroup/NakedObjectsFramework#210]
e.g. Customer_Functions.CustomerType

(test that member order is correct here)

**Comment by richardpawson** [comment]
1)	I did successfully use DisplayAsProperty, for a reference type, on the client.
2)	The [MemberOrder] did not result in correct positioning, which is not entirely surprising.  I, incidentally, confirmed your memory that while the specified member order value (e.g. 17) is included in the RO representation of any property/action (under extensions), the server has pre-sorted the properties/actions based on the member order, and the client uses this ordering rather than the actual MemberOrder values to create the presentation.  We’ll park the issue for now.

Meantime …

While looking at my perceived need for DisplayAsProperty to work on value types, I discovered, beleatedly, that there is a very elegant syntax in C# 9 for derived, read-only properties i.e.: like an instance function definition but with no brackets: e.g.:

public decimal TotalValue => My_Functions.CalculateTotalValue(this);

I have used this now in multiple places in AW, and ended up replacing all my intended uses of DisplayAsProperty with this syntax.

However, the work on DisplayAsProperty is not wasted. It will be needed for two cases:

1)	For a derived reference type or collection that needs to be retrieved from the database (via IContext) rather than from other properties.
2)	For properties/collections to be contributed to abstract/interface types (this is the DSP’s upcoming requirement for NOF12)

However, having spiked the idea I suggest we now park it – as you said it needs more specification.
